wire/netaddress.go

```diff
@@ -90,12 +83,13 @@ func NewNetAddress(addr *net.TCPAddr, services ServiceFlag) *NetAddress {
 func readNetAddress(r io.Reader, pver uint32, na *NetAddress, ts bool) error {
 	var services ServiceFlag
 	var ip [16]byte
+	timestamp := na.Timestamp
```

Comment:
I would prefer this section be left the way it is.  The reason is that doing it this way forces another allocation for every single `NetAddress` that is reads.  There was a recent commit that intentionally removed this allocation (and eventually it should do the same for the other two fields as well).

Before this PR:
```
$ go test -run=NONE -bench=BenchmarkDecodeAddr -benchmem
BenchmarkDecodeAddr-6            300           6304134 ns/op        97728 B/op    2002 allocs/op
```
After:

```
$ go test -run=NONE -bench=BenchmarkDecodeAddr -benchmem
BenchmarkDecodeAddr-6            100          10371317 ns/op       129756 B/op    3002 allocs/op
```

We can see the additional allocation is indeed this line here:

```
$ go test -run=NONE -bench=BenchmarkDecodeAddr -memprofile=mem.out -memprofilerate=1
$ go tool pprof -alloc_objects ./wire.test ./mem.out
Entering interactive mode (type "help" for commands)
(pprof) list readNetAddress
Total: 263037
ROUTINE ======================== github.com/btcsuite/btcd/wire.readNetAddress in D:/Projects/mygo/src/github.com/btcsuite/btcd/wire/netaddress.go
    252502     252502 (flat, cum) 95.99% of Total
         .          .     79:
         .          .     80:// readNetAddress reads an encoded NetAddress from r depending on the protocol
         .          .     81:// version and whether or not the timestamp is included per ts.  Some messages
         .          .     82:// like version do not include the timestamp.
         .          .     83:func readNetAddress(r io.Reader, pver uint32, na *NetAddress, ts bool) error {
     50502      50502     84:   var services ServiceFlag
    101000     101000     85:   var ip [16]byte
    101000     101000     86:   timestamp := na.Timestamp
         .          .     87:
         .          .     88:   // NOTE: The bitcoin protocol uses a uint32 for the timestamp so it will
         .          .     89:   // stop working somewhere around 2106.  Also timestamp wasn't added until
         .          .     90:   // protocol version >= NetAddressTimeVersion
         .          .     91:   if ts && pver >= NetAddressTimeVersion {
(pprof)